### PR TITLE
Submodule conversion

### PR DESCRIPTION
**Pull request by grod220**
Closes https://github.com/prax-wallet/registry/issues/5
Closes https://github.com/prax-wallet/registry/issues/85

The [initial motivation](https://github.com/prax-wallet/registry/issues/5#issuecomment-2049316786) to use http requests has proven to be more finicky than hoped for. It introduces failure possibility during runtime when the submodule strategy pulls those issues in only after the cosmos registry dep is updated. This will allow us to have deterministic builds for the current commit and worry less about the `cargo run` command failing when we are trying to add another chain id. The downside is that it requires 150mb download when pulling down for the first time.

### CODE REVIEW

File: .github/workflows/rust-checks.yml [hunk 1]
```diff
@@ -52,3 +58,24 @@ jobs:
...
+          toolchain: stable
+          override: true
+      - name: Generate registry
+        run: cargo run
+        env:
+          RUST_LOG: info
+        working-directory: tools/compiler
+      - name: Check for Uncommitted Changes
+        run: git diff --exit-code
+        working-directory: tools/compiler
```
Comment (grod220) [inline .github/workflows/rust-checks.yml:81]: Allows us now to ensure folks are properly running the script and not doing manual edits

File: npm/package.json
Comment (grod220) [inline npm/package.json:1]: Cutting a new release for consumers using bundled registries

File: npm/src/json.ts [hunk 2]
```diff
@@ -1,4 +1,4 @@
-import * as Deimos8 from '../../registry/chains/penumbra-testnet-deimos-8.json';
+import * as Deimos8 from '../../registry/chains/penumbra-testnet-deimos-8-x6de97e39.json';
```
Comment (grod220) [inline npm/src/json.ts:1]: It looks like the `penumbra-testnet-deimos-8.json` has been removed from the inputs. To pass the test, these need to be updated.

File: registry/chains/penumbra-1.json
Comment (grod220) [inline registry/chains/penumbra-1.json:1]: Diff due to pulling from the latest commit from the cosmos registry

File: registry/chains/penumbra-testnet-deimos-8.json
Comment (grod220) [inline registry/chains/penumbra-testnet-deimos-8.json:1]: I don't think anyone is depending on this (?). But the input that generates this seems to have been deleted a while ago.

File: npm/src/json.ts (hunk 2 above)
Comment (conorsch) [reply npm/src/json.ts:1]: Related to https://github.com/prax-wallet/registry/pull/73, last time L and I were debugging. IIRC we were surprised to see that the relevant codepaths were dropping the `-x6de97e39` and trying to import the suffix-less JSON file, which is _not_ what I wanted. I suspect this is still an issue, but we can follow up and file if testing surfaces it again.

Comment (grod220) [reply npm/src/json.ts:1]: Discussed offline: made an update to the chain id swapping logic to only do so if the exact match fails

**Review by conorsch — APPROVED** [review]
Long overdue, very usable, thank you!

**Pull request merged**